Add checklist (#21)
filecoin-project/f3-activation-contract: checklist.md

## F3 Activation Procedure (FRC-0099)

> This document outlines the procedure for activating F3 on the Filecoin network, adhering to FRC-0099. The process involves a multi-step approach, leveraging a smart contract to ensure transparency and community consensus.

Reviewer: ```suggestion
### Purpose
This document outlines the procedure for activating F3 on the Filecoin network, adhering to [FRC-0099](https://github.com/filecoin-project/FIPs/blob/master/FRCs/frc-0099.md). The process involves a multi-step approach, leveraging a smart contract to ensure transparency and community consensus.

This checklist will be copied into the ["tracking issue"](https://github.com/filecoin-project/f3-activation-contract/issues/22) before we start the activation procedure so that the current status can be tracked and commented upon.
```

> 

Reviewer: ```suggestion

### Meta
* Mainnet contract address: [0xA19080A1Bcb82Bb61bcb9691EC94653Eb5315716](https://filecoin.blockscout.com/address/0xA19080A1Bcb82Bb61bcb9691EC94653Eb5315716?tab=contract)
* Multisig owning contract: [0x53bd89Ff2Ff97541f42ACC3AFC0C0030e7410422](https://safe.filecoin.io/settings/setup?safe=filecoin:0x53bd89Ff2Ff97541f42ACC3AFC0C0030e7410422)

### Steps
```

Reviewer: I'm thinking to pull these key parameter up top

1.  **Parameter Selection and Bootstrap Epoch Determination**
    *   Goal: Establish the "good parameter set" for F3 and the initial epoch for its activation.

>     *   Action: Analyze nv25 passive testing results. Consider network performance, stability, and community input.

Reviewer: ```suggestion
    *   Action: Analyze nv25 passive testing results. Consider network performance, stability, and community input.
    * Status: not started
```

I'm thinking we add a status section to each step that we'll update as we make progress through the checklist

>     *   Output: A finalized set of F3 parameters and a proposed bootstrap epoch.

Reviewer: ```suggestion
    *   Output: A finalized set of F3 parameters and a proposed bootstrap epoch are posted in the "tracking issue".
```

> 2.  **Manifest Definition and Storage**

Reviewer: ```suggestion
2.  **Manifest Definition, Storage, and Communication**
```

I am thinking we do this via PR which then also pulls in the communication

>     *   Goal: Formally define the parameters and bootstrap epoch for distribution.

Reviewer: ```suggestion
    *   Goals: 
         * Formally define the parameters and bootstrap epoch for distribution.
         * Ensure all stakeholders are informed and aligned.
```

    *   Action:

>         *   Create a manifest file (e.g., JSON) that clearly defines all F3 parameters and the selected bootstrap epoch.

Reviewer: ```suggestion
        * Create a PR against [mainnet-manifest.json](https://github.com/filecoin-project/f3-activation-contract/blob/master/tasks/mainnet-manifest.json) that clearly defines all F3 parameters and the selected bootstrap epoch.
        * Request review from Lotus (@masih), Forest (@LesnyRumcajs), and Venus (@simlecode).
        * Post a link to the PR in #fil-implementers.
        * Post a link to the PR in the "tracking issue".
```

        *   Commit the manifest file into the `f3-activation-contract` repository.

>     *   Output: A verifiable manifest file stored in the repository.

Reviewer: ```suggestion
    *   Status: not started
    *   Output: A verifiable manifest file stored in the repository.
```

    *   **Manifest Link:** \[*Link to manifest file in repository*]

> 3.  **Communication and Coordination**

Reviewer: I think with the modification above that this step can be removed?

Author: I think I agree

    *   Goal: Ensure all stakeholders are informed and aligned.
    *   Action:
        *   Communicate the proposed F3 parameters, bootstrap epoch, and rationale to Lotus, Forest, and Venus implementation teams.
        *   Establish the expected encoding scheme of the manifest so that it is the same across implementations.
        *   Share the manifest file and the address of the smart contract.
    *   **Smart Contract Address:** \[*F3 Activation Smart Contract Address*]

> 4.  **Smart Contract Interaction using SAFE Transaction Builder**

Reviewer: Just curious why you're thinking to use the SAFE Transaction Builder UI vs. do this in the CLI so it can be reviewed now and just copy/pasted to execute.

Author: SAFE Transaction Builder is used to propose the transaction within SAFE wallet, with the inputs coming from CLI.

Reviewer: Got it.  I was wondering about using something like https://github.com/safe-global/safe-cli .  I quickly tried this but ran into python environment / dependency issues and didn't want to sink much into it.  I think we are fine to defer, but I think it's an "area of improvement" if this contract-based rollout strategy is used in the future.

    *   Goal: Prepare and execute the transaction that sets the parameters in the smart contract.
    *   Action:
        *   Use the SAFE Transaction Builder to craft a transaction that interacts with the F3 activation smart contract.
        *   The transaction should call the function on the smart contract that sets the F3 parameters and bootstrap epoch.

>         *   Verify that the encoded manifest is correctly formatted and points to the parameters.

Reviewer: If there is any output that can be captured, add a step to post it to the "tracking issue"?

Reviewer: Ack, makes sense.  For now in https://github.com/filecoin-project/f3-activation-contract/pull/21/commits/87613d6ab4f51c6482ae440eebe944b024b43a53 I added an output of the "screenshot of the Safe Transaction Builder".  Resolving.

>     *   **Multi-Sig Address:** \[*Address of the Multi-Sig contract*]

Reviewer: ```suggestion
    * Status: not started
    * Output: transaction 
```

Or I guess there isn't a transaction at this point because we don't have the signatures.  Is there anything we can capture from this step?

5.  **Multi-Sig Approval**
    *   Goal: Achieve consensus among implementation partners.
    *   Action:

>         *   Implementation partners (Lotus, Forest, Venus) should review the proposed transaction in the SAFE Transaction Builder.

Reviewer: Are you thinking this is better to do in the UI than via the CLI?

Author: ```suggestion
        *   Implementation partners (Lotus, Forest, Venus) should review the proposed transaction via the CLI.
```

        *   Each partner uses their key to sign the transaction, indicating their approval of the F3 parameters and bootstrap epoch.

6.  **Transaction Publication and Execution**
    *   Goal: Activate F3 on the network.
    *   Action:
        *   Once the required number of signatures is collected, publish the signed transaction to the Filecoin network.

>         *   The smart contract will execute the transaction, setting the F3 parameters and scheduling the bootstrap epoch.

Reviewer: ```suggestion
        *   The smart contract will execute the transaction, setting the F3 parameters and scheduling the bootstrap epoch.
        * Post the transaction hash to the "tracking issue".
    * Status: not started
```

    *   **Transaction Hash:** \[*Transaction Hash of the Activation Transaction*]

7.  **Verification and Monitoring**
    *   Goal: Confirm successful activation and ongoing stability.
    *   Action:

>         *   After the bootstrap epoch, verify that F3 is active on the network by monitoring relevant metrics .

Reviewer: ```suggestion
        *   After the bootstrap epoch, verify that F3 is active on the network by monitoring relevant metrics.
```

>         *   Continuously monitor network performance and adjust parameters as needed, following the same process.

Reviewer: > adjust parameters as needed, following the same process.

How does this work?  I think this could only be used once.

Also, I'd like to have this step be something discrete that we can mark as done.  (Agreed monitoring will need to continue).  Basically, I assume once we confirm F3 is active and working on mainnet we'd  call this step (and thus the "tracking issue") done.

Author: Agreed, we don't really get to adjust further in the network version. LLM I used for partial re-writes has gotten a bit carried away and I missed it.

Reviewer: I adjusted this in https://github.com/filecoin-project/f3-activation-contract/pull/21/commits/87613d6ab4f51c6482ae440eebe944b024b43a53 .  Resolving.
